shortfin/python/shortfin_apps/sd/components/config_struct.py

```diff
@@ -59,11 +65,13 @@ class ModelParams:
     # Classifer free guidance mode. If set to false, only positive prompts will matter.
     cfg_mode = True
 
-    # DTypes (basically defaults):
+    # DTypes (not necessarily weights precision):
     clip_dtype: sfnp.DType = sfnp.float16
     unet_dtype: sfnp.DType = sfnp.float16
     vae_dtype: sfnp.DType = sfnp.float16
 
+    use_i8_punet: bool = False
```

Comment:
really don't like this but the module expects fp16 I/O...

Reply:
Maybe a specific place for IO dtype is in order. One (`*_dtype`) is used for instructing device array creation, and the other (`use_i8_punet`) is used when inferring artifact names.

Reply:
Drop by comment, I agree with the above. We have multiple `punet` models to support like `int8` and `fp8`, so it would be better to keep them separate